src/Yoti.Auth/Activity.cs

```diff
@@ -36,23 +36,57 @@ public ActivityDetails HandleSuccessfulResponse(AsymmetricCipherKeyPair keyPair,
             {
                 return new ActivityDetails(ActivityOutcome.SharingFailure);
             }
-            else
-            {
-                ReceiptDO receipt = parsedResponse.Receipt;
 
-                AttrpubapiV1.AttributeList attributes = CryptoEngine.DecryptCurrentUserReceipt(
-                    parsedResponse.Receipt.wrapped_receipt_key,
-                    parsedResponse.Receipt.other_party_profile_content,
-                    keyPair);
+            ReceiptDO receipt = parsedResponse.Receipt;
 
-                _yotiUserProfile.Id = parsedResponse.Receipt.remember_me_id;
-                _yotiProfile.Id = parsedResponse.Receipt.remember_me_id;
-                string receiptId = parsedResponse.Receipt.receipt_id;
+            ParseOtherPartyProfileContent(keyPair, parsedResponse);
+            Dictionary<string, BaseAttribute> applicationProfileAttributes = ParseApplicationProfileContent(keyPair, parsedResponse);
 
-                AddAttributesToProfile(attributes);
+            ApplicationProfile applicationProfile = new ApplicationProfile(applicationProfileAttributes);
+
+            DateTime? timestamp = null;
+            if (receipt.timestamp != null)
+            {
+                if (DateTime.TryParseExact(receipt.timestamp, "yyyy-MM-ddTHH:mm:ssZ", CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTime parsedDate))
+                {
+                    timestamp = parsedDate;
+                }
+            }
 
-                return new ActivityDetails(_yotiUserProfile, _yotiProfile, receiptId, ActivityOutcome.Success);
+            return new ActivityDetails(parsedResponse.Receipt.remember_me_id, timestamp, _yotiUserProfile, _yotiProfile, applicationProfile, parsedResponse.Receipt.receipt_id, ActivityOutcome.Success);
+        }
+
+        private void ParseOtherPartyProfileContent(AsymmetricCipherKeyPair keyPair, ProfileDO parsedResponse)
+        {
+            AttrpubapiV1.AttributeList profileAttributes = CryptoEngine.DecryptCurrentUserReceipt(
```

Comment:
You might want to eliminate decryption redundancy for both OtherProfileContent and ProfileContent, by moving this into one place AttributesListConverter::class.
Again this is a suggestion.

Reply:
I've added this to https://lampkicking.atlassian.net/browse/SDK-587 now